apps/server/src/modules/tool/service/context-external-tool.service.spec.ts

```diff
@@ -77,4 +80,44 @@ describe('ContextExternalToolService', () => {
 			});
 		});
 	});
+
+	describe('getContextExternalToolById is called', () => {
+		describe('when contextExternalToolId is given', () => {
+			const setup = () => {
+				const contextExternalTool: ContextExternalToolDO = contextExternalToolDOFactory.build();
+
+				return {
+					contextExternalToolId: contextExternalTool.id as string,
+				};
+			};
+
+			it('should call getContextExternalToolById()', async () => {
+				const { contextExternalToolId } = setup();
+
+				await service.getContextExternalToolById(contextExternalToolId);
+
+				expect(contextExternalToolRepo.findById).toHaveBeenCalledWith(contextExternalToolId);
+			});
+		});
+	});
+
+	describe('deleteContextExternalTool is called', () => {
+		describe('when contextExternalToolId is given', () => {
+			const setup = () => {
+				const contextExternalTool: ContextExternalToolDO = contextExternalToolDOFactory.build();
+
+				return {
+					contextExternalTool,
+				};
+			};
+
+			it('should call deleteContextExternalTool()', async () => {
```

Comment:
Should be something like "should call delete on repo". 

Refering to the above comment: Here the call to the repo has a side effect, i.e. makes changes in the database, and you should test for it like you do.